app/Services/Utils/EmailExcerptService.php

```diff
@@ -0,0 +1,38 @@
+<?php namespace App\Services\utils;
+
+/**
+ * Copyright 2022 OpenStack Foundation
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+/**
+ * Class EmailExcerptService
+ * @package App\Services\utils
+ */
+final class EmailExcerptService implements IEmailExcerptService
+{
+    private $report = [];
```

Comment:
please add type definitions and php docs comments

Reply:
done